Periodically prepare a combined custom + system CA bundle

bbockelm opened the pull request:
Both XRootD and Prometheus can't use our custom transport object, which combines the system CA pool with the custom CA.  This causes various failures unless you disable TLS validation (which we want to avoid doing).

This adds a helper function that periodically (re)creates the CA bundle. These bundles are made available to other functions at runtime.

NOTE: Neither Mac nor Windows have a standard system bundle.  On those platforms, we only create the bundle if the custom CA is present.

bbockelm commented:
@turetske - this is the PR I mentioned at standup.  It creates a "Pelican CA bundle" for use by components that can't utilize our custom transport inside the `pelican` process.

turetske reviewed (changes requested):
Small spelling changes.

Just to make sure I understand what's occurring in the code: This is periodically taking our custom certs and the system certs, combining them into one file, and then writing that file out to a common location for prometheus and xrootd (in this case) to read?

bbockelm commented on utils/ca_utils.go:
```diff
@@ -0,0 +1,225 @@
+	return uniq, nil
+}
+
+// isSameDirSymlink reports whether fi in dir is a symlink with a
```
Done!

bbockelm commented on utils/ca_utils.go:
```diff
@@ -0,0 +1,225 @@
+)
+
+// Write out all the trusted CAs as a CA bundle on disk.  This is useful
+// for components that do not use go's trusted CA stroe
```
Done!

bbockelm commented:
Overriding branch  protections since I addressed @turetske's comments (and want to build on top of this in order to make the cache easier to test).  The test failure appears to be spurious and fixed elsewhere.
